src/varint.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "common.h"
+#include "varint.h"
+
+uintmax_t decode_varint(const unsigned char **bufp)
```

Comment:
Our convention is generally to not have "in/out" params like this.  Generally we prefer to take the `char *` and return the size used rather than also modifying it.